test/unit-test/gtclang/Parsing/TestParsing.cpp

```diff
@@ -0,0 +1,175 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                         _       _
+//                        | |     | |
+//                    __ _| |_ ___| | __ _ _ __   __ _
+//                   / _` | __/ __| |/ _` | '_ \ / _` |
+//                  | (_| | || (__| | (_| | | | | (_| |
+//                   \__, |\__\___|_|\__,_|_| |_|\__, | - GridTools Clang DSL
+//                    __/ |                       __/ |
+//                   |___/                       |___/
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/CodeGen/CXXUtil.h"
+#include "dawn/SIR/SIR.h"
+#include "gtclang/Support/ParsingHeader.h"
+#include "gtclang/Unittest/GTClang.h"
+#include "gtclang/Unittest/UnittestEnvironment.h"
+#include <fstream>
+#include <gtest/gtest.h>
+#include <map>
+
+using namespace dawn;
+using namespace gtclang;
+
+namespace {
+
+struct NestableFunctions : public codegen::MemberFunction {
+  NestableFunctions(const Twine& type, const Twine& name, std::stringstream& s, int il = 0)
+      : MemberFunction(type, name, s, il) {}
+
+  NestableFunctions addFunction(const Twine& returntype, const Twine& name) {
+    NestableFunctions nf(returntype, name, ss(), IndentLevel + 1);
+    return nf;
+  }
+
+  NestableFunctions addVerticalRegion(const Twine& min, const Twine& max) {
+    NestableFunctions nf(Twine::createNull(), "vertical_region", ss(), IndentLevel + 1);
+    nf.addArg(min).addArg(max);
+    nf.startBody();
+    return nf;
+  }
+
+  NestableFunctions addIfStatement(const Twine& ifCondition) {
+    NestableFunctions ifStatement(Twine::createNull(), "if", ss(), IndentLevel + 1);
+    ifStatement.addArg(ifCondition);
+    ifStatement.startBody();
+    return ifStatement;
+  }
+};
+
+struct VerticalRegion : public NestableFunctions {
+  VerticalRegion(const Twine& name, std::stringstream& s, int il = 0)
+      : NestableFunctions(Twine::createNull(), name, s, il) {}
+};
+
+struct DawnObject : public codegen::Structure {
```

Comment:
DawnObject? It's not clear to me what this generates? I would avoid Object in a class name (better use XYBase or something)